core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -1032,7 +1087,10 @@ private synchronized void cancelConflictResolutionPhase(CacheTopology resolution
                      rootCause = cause;
                   }
 
-                  if (trace) log.tracef("Cache %s cancelling conflict resolution due to root cause t=%s", cacheName, rootCause);
+                  // TODO When CR fails because a node left the cluster, the new CR can start before we cancel the old one
```

Comment:
Maybe it's because the queueing happens outside the `LimitedExecutor`, but I got this in the log:

```
11:55:23,231 DEBUG (transport-thread-test-NodeA-p4-t5:[ViewHandling]) [ClusterCacheStatus] Cache ___defaultcache restarting conflict resolution with topology CacheTopology{id=21, phase=CONFLICT_RESOLUTION, rebalanceId=7, currentCH=DefaultConsistentHash{ns=256, owners = (3)[test-NodeD-63185: 256+0, test-NodeA-1720: 0+256, test-NodeB-44683: 0+256]}, pendingCH=null, unionCH=null, actualMembers=[test-NodeD-63185, test-NodeA-1720, test-NodeB-44683], persistentUUIDs=[c42fd10e-0898-4566-976b-363cdee3e54d, 88ae2d00-e2e3-44bd-bf5e-634f2ae2e16f, 430ac0c2-59f6-439d-8711-d721ef612193]}
11:55:23,233 DEBUG (transport-thread-test-NodeA-p4-t5:[ViewHandling]) [ClusterCacheStatus] Cache ___defaultcache queueing conflict resolution with members [test-NodeD-63185, test-NodeA-1720, test-NodeB-44683]
11:55:23,234 INFO  (transport-thread-test-NodeA-p4-t5:[ViewHandling]) [CLUSTER] [Context=___defaultcache] ISPN100012: Starting conflict resolution with members [test-NodeD-63185, test-NodeA-1720, test-NodeB-44683], topology id 21
11:55:23,247 INFO  (stateTransferExecutor-thread-test-NodeA-p5-t3:[ConflictManager-]) [CLUSTER] [Context=___defaultcache] ISPN100014: Failed conflict resolution with members [test-NodeC-63433, test-NodeD-63185, test-NodeA-1720, test-NodeB-44683], topology id 20: ISPN000400: Node test-NodeC-63433 was suspected
```

Reply:
I agree it's because the queuing is outside of the executor. `ClusterCacheStatus::queueConflictResolution` can be called before the old CR finishes resulting in `ISPN100012: Starting...` appearing before `ISPN100014: Failed...`. We can confirm this with trace logging, so if `Cache %s attempting to resolve conflicts` appears before `ISPN100014` we have CR overlapping.